art/estimators/object_detection/tensorflow_faster_rcnn.py

```diff
@@ -0,0 +1,530 @@
+            ) for i in range(images.shape[0])
+        ]
+
+        # Create placeholders for groundtruth weights
+        self._groundtruth_weights_list: List["Tensor"]
+        self._groundtruth_weights_list = [
+            tf.placeholder(
+                dtype=tf.float32,
+                shape=(None,),
+                name="groundtruth_weights_{}".format(i)
+            ) for i in range(images.shape[0])
+        ]
+
+        # Load model
+        if model is None:
+            # If model is None, then we need to have parameters filename and url to download, extract and load the
+            # object detection model
+            if filename is None or url is None:
+                filename, url = (
+                    'faster_rcnn_inception_v2_coco_2017_11_08',
+                    'http://download.tensorflow.org/models/object_detection/'
+                    'faster_rcnn_inception_v2_coco_2017_11_08.tar.gz'
+                )
+
+            self._predictions, self._losses, self._detections = self._load_model(
+                filename=filename,
+                url=url,
+                obj_detection_model=None,
+                images=images,
+                is_training=is_training,
+                groundtruth_boxes_list=self._groundtruth_boxes_list,
+                groundtruth_classes_list=self._groundtruth_classes_list,
+                groundtruth_weights_list=self._groundtruth_weights_list
+            )
+
+        else:
+            self._predictions, self._losses, self._detections = self._load_model(
+                filename=None,
+                url=None,
+                obj_detection_model=model,
+                images=images,
+                is_training=is_training,
+                groundtruth_boxes_list=self._groundtruth_boxes_list,
+                groundtruth_classes_list=self._groundtruth_classes_list,
+                groundtruth_weights_list=self._groundtruth_weights_list
+            )
+
+        # Save new attributes
+        self._input_shape = images.shape.as_list()[1:]
+        self._is_training: bool = is_training
+        self._images: Optional["Tensor"] = images
+        self._attack_losses: Tuple[str, ...] = attack_losses
+
+        # Assign session
+        if sess is None:
+            logger.warning("A session cannot be None, create a new session.")
+            self._sess: "Session" = tf.Session()
+        else:
+            self._sess: "Session" = sess
+
+        # Initialize variables
+        self._sess.run(tf.global_variables_initializer())
+        self._sess.run(tf.local_variables_initializer())
+
+    @staticmethod
+    def _load_model(
+        filename: Optional[str] = None,
+        url: Optional[str] = None,
+        obj_detection_model: Optional["FasterRCNNMetaArch"] = None,
+        images: Optional["Tensor"] = None,
+        is_training: bool = False,
+        groundtruth_boxes_list: Optional[List["Tensor"]] = None,
+        groundtruth_classes_list: Optional[List["Tensor"]] = None,
```

Comment:
I think `groundtruth_classes_list` cannot be None. Maybe we should remove the default here?
```suggestion
        groundtruth_classes_list: List["Tensor"],
```

Reply:
When you load the faster rcnn model for testing (is_training=False), you don't need to provide those three groundtruths, so they can be None in this case. The TensorflowFasterRCNN model is implemented in such a way that it allows ART users to use it in more cases than just for ShapeShifter where we indeed only need is_training=True.